src/ddmd/mars.d

```diff
@@ -245,6 +245,29 @@ extern (C++) void genCmain(Scope* sc)
     rootHasMain = sc._module;
 }
 
+private void semantic3OnDependencies(Module m, StringTable *importArray)
```

Comment:
No, they hand-maintain a copy of much of the code in `mars.d`.  Putting this function in `mars.d` when it contains only backend-independent functionality will force them to do the same with this code.  Besides, this function logically belongs with the other module code, doesn't it?  It's not like putting it here allows mars not to depend on module.